src/Internal/Payment/State/InitialState.php

```diff
@@ -8,18 +8,47 @@
 namespace WCPay\Internal\Payment\State;
 
 use WCPay\Internal\Payment\Exception\StateTransitionException;
+use WCPay\Exceptions\Order_Not_Found_Exception;
+use WCPay\Internal\Service\OrderService;
 
 /**
  * Initial state, representing a freshly created payment.
  */
 class InitialState extends PaymentState {
+	/**
+	 * Order service.
+	 *
+	 * @var OrderService
+	 */
+	private $order_service;
+
+	/**
+	 * Class constructor, only meant for storing dependencies.
+	 *
+	 * @param StateFactory $state_factory Factory for payment states.
+	 * @param OrderService $order_service Service for order-related actions.
+	 */
+	public function __construct(
+		StateFactory $state_factory,
+		OrderService $order_service
+	) {
+		parent::__construct( $state_factory );
+
+		$this->order_service = $order_service;
+	}
+
 	/**
 	 * Initialtes the payment process.
 	 *
 	 * @return CompletedState The next state.
-	 * @throws StateTransitionException In case the completed state could not be initialized.
+	 * @throws StateTransitionException Completed state could not be initialized.
+	 * @throws Order_Not_Found_Exception Order could not be found.
 	 */
 	public function process() {
+		$order_id       = $this->get_context()->get_order_id();
```

Comment:
Wouldn't be wise here to store the whole instance of WP_Order and prevent calling the order from the database on every step?

Reply:
I'm 50/50 here. @marcinbot can join the conversation, as this is done per his suggestion. IMO it makes sense to keep the `PaymentContext` simple and easily serializable, but at the same time, now we need to mock a ton of `get_order()` calls within simple methods, and there are `Order_Not_Found` exceptions that need to be documented throughout states and services. Providing an order in advance would eliminate that,

Reply:
I think we could keep it this way for now, and only consider moving back to an object in case it gets too overwhelming.